core/src/main/java/org/infinispan/protostream/impl/TagWriterImpl.java

```diff
@@ -66,9 +67,13 @@ public static TagWriterImpl newNestedInstance(ProtobufTagMarshaller.WriteContext
    }
 
    public static TagWriterImpl newInstance(ImmutableSerializationContext serCtx, OutputStream output) {
-      return new TagWriterImpl((SerializationContextImpl) serCtx, new OutputStreamEncoder(output, ProtobufUtil.DEFAULT_STREAM_BUFFER_SIZE));
+      return new TagWriterImpl((SerializationContextImpl) serCtx, new OutputStreamNoBufferEncoder(output));
```

Comment:
vent: I hate this castings!
nitpick, it can use the new method to instantiate the new `TagWriterImpl`
 ```suggestion
      return newInstance(serCtx, new OutputStreamNoBufferEncoder(output));
```

Reply:
offtopic for this line, `newNestedInstance` can be removed/deprecated and should use `subWriter()`?

Reply:
> vent: I hate this castings!

I totally agree, I hate the cast but that is how it was. 

> nitpick, it can use the new method to instantiate the new `TagWriterImpl`

I will update them all to use the new method so the cast is only in one spot.